src/main/webapp/app/instructor-dashboard/instructor-dashboard-result-dialog.controller.js

```diff
@@ -5,24 +5,35 @@
         .module('artemisApp')
         .controller('InstructorDashboardResultDialogController', InstructorDashboardResultDialogController);
 
-    InstructorDashboardResultDialogController.$inject = ['$timeout', '$scope', '$stateParams', '$uibModalInstance', 'entity', 'participationEntity', 'Result', 'AlertService'];
+    InstructorDashboardResultDialogController.$inject = ['$timeout', '$scope', '$stateParams', '$uibModalInstance', 'entity', 'participationEntity', 'Result', 'AlertService', 'Feedback'];
 
-    function InstructorDashboardResultDialogController($timeout, $scope, $stateParams, $uibModalInstance, entity, participationEntity, Result, AlertService) {
+    function InstructorDashboardResultDialogController($timeout, $scope, $stateParams, $uibModalInstance, entity, participationEntity, Result, AlertService, Feedback) {
         var vm = this;
 
         vm.result = entity;
         vm.clear = clear;
         vm.datePickerOpenStatus = {};
         vm.openCalendar = openCalendar;
         vm.save = save;
+        vm.openForSubmission = openForSubmission;
+        vm.addFeedbackClicked = false;
+        vm.feedbackIndices = [0];
+        vm.feedbacks = [];
+        vm.pushFeedback = pushFeedback;
+        vm.popFeedback = popFeedback;
 
         if(participationEntity) {
             entity.participation = participationEntity;
         } else {
             clear();
         }
 
-
+        function openForSubmission() {
+            if(vm.result.participation.exercise.dueDate == null || Date.now() < Date.parse(vm.result.participation.exercise.dueDate)) {
```

Comment:
please do not use Date.now(). The time of the client might be different from the time on the server. Make sure to compute this property "openForSubmission" on the server and send it e.g. using a temporary attribute only stored in JSON, but not in the database.

Reply:
Done.